### PR TITLE
More readable 403s

### PR DESCRIPTION
**Pull request by eric-maynard**
# Description

Currently, when a request is rejected users see an error like this:

<code>org.apache.iceberg.exceptions.ForbiddenException: Forbidden: Principal 'quickstart_user' with activated PrincipalRoles '[]' and activated ids '[6, 7]' is not authorized for op CREATE_NAMESPACE</code>

With this change, users will see something more like the following:
<code>org.apache.iceberg.exceptions.ForbiddenException: Forbidden: Principal 'quickstart_user' with activated PrincipalRoles '[]' and activated grants via '[quickstart_catalog_role, quickstart_user_role]' is not authorized for op CREATE_NAMESPACE</code>

### CODE REVIEW

File: polaris-core/src/main/java/org/apache/polaris/core/auth/PolarisAuthorizer.java [hunk 1]
```diff
@@ -629,4 +634,8 @@ public boolean hasTransitivePrivilege(
         resolvedPath);
     return false;
   }
+
+  private Set<String> granteeIdsToNames(Set<Long> granteeIds) {
```
Comment (ebyhr) [inline polaris-core/src/main/java/org/apache/polaris/core/auth/PolarisAuthorizer.java:638]: Can we remove this method?

**Review by ebyhr — APPROVED** [review]

File: polaris-core/src/main/java/org/apache/polaris/core/auth/PolarisAuthorizer.java [hunk 2]
```diff
@@ -514,12 +517,12 @@ public void authorizeOrThrow(
           "Principal '%s' is not authorized for op %s due to PRINCIPAL_CREDENTIAL_ROTATION_REQUIRED_STATE",
           authenticatedPrincipal.getName(), authzOp);
     } else if (!isAuthorized(
-        authenticatedPrincipal, activatedGranteeIds, authzOp, targets, secondaries)) {
+        authenticatedPrincipal, activatedEntities, authzOp, targets, secondaries)) {
       throw new ForbiddenException(
-          "Principal '%s' with activated PrincipalRoles '%s' and activated ids '%s' is not authorized for op %s",
+          "Principal '%s' with activated PrincipalRoles '%s' and activated grants via '%s' is not authorized for op %s",
```
Comment (flyrain) [inline polaris-core/src/main/java/org/apache/polaris/core/auth/PolarisAuthorizer.java:522]: Minor: Not sure if this is clearer? `grants via` -> `grant on`

**Review by flyrain — APPROVED** [review]

File: polaris-core/src/main/java/org/apache/polaris/core/auth/PolarisAuthorizer.java (hunk 2 above)
Comment (eric-maynard) [reply polaris-core/src/main/java/org/apache/polaris/core/auth/PolarisAuthorizer.java:522]: I think what this message is trying to say is that you have grants _through_ / _via_ these roles. So you may have a grant on table X through your principal role A and catalog role B but you fail when you try to read table Y. This message will say `...and activated grants via [A, B]`

Comment (eric-maynard) [reply polaris-core/src/main/java/org/apache/polaris/core/auth/PolarisAuthorizer.java:522]: I guess you also have grants _on_ A and B but they are not privilege grants, they are role grants.

Comment (flyrain) [reply polaris-core/src/main/java/org/apache/polaris/core/auth/PolarisAuthorizer.java:522]: My bad. I thought they table/namespace entities.

Comment (eric-maynard) [reply polaris-core/src/main/java/org/apache/polaris/core/auth/PolarisAuthorizer.java:522]: Ah no they are roles. But I am still not sure if this verbiage is totally correct, maybe you are right

Comment (flyrain) [reply polaris-core/src/main/java/org/apache/polaris/core/auth/PolarisAuthorizer.java:522]: The `via` makes sense to me if they are roles.

**Pull request merged**